src/Controls/src/Core/Span.cs

```diff
@@ -159,7 +159,7 @@ void IFontElement.OnFontSizeChanged(double oldValue, double newValue)
 		}
 
 		double IFontElement.FontSizeDefaultValueCreator() =>
-			this.GetDefaultFontSize();
+			double.NaN;
```

Comment:
This is the only "breaking" change, but anyone using this value without it being set by the user is getting a wrong value. I could use `0`?

Reply:
It is, sort of. The method returns 14 if the span is attached and 0 if not. The issue with these default value functions is that it sets the value as if the user set it.

So the OG issue is that the label font was set to a big size, and then the font remains small because by the time the label is rendered the framework overrides the big with the default of 14.

So if you do:

```cs
// in ctor
var firstSpan = new Span();
var firstSize = span.FontSize; // <- 0

var secondSpan = new Span();
var label = new Label {
    FontSIze = 100,
    FormattedText = new FormattedString { secondSpan }
};

// after rendering
var secondSize = secondSpan.FontSize; // <- 14
```

In both cases, you did not set a font size on the span, and in the second you actually set a size on the label. And that size is not used.

The NaN is a way to tell the rest of the framework that no size has been set, so keep looking.